ledger/tracker.go

```diff
@@ -322,16 +322,14 @@ func (tr *trackerRegistry) committedUpTo(rnd basics.Round) basics.Round {
 }
 
 func (tr *trackerRegistry) scheduleCommit(blockqRound, maxLookback basics.Round) {
-	tr.mu.RLock()
-	dbRound := tr.dbRound
-	tr.mu.RUnlock()
-
 	dcc := &deferredCommitContext{
 		deferredCommitRange: deferredCommitRange{
 			lookback: maxLookback,
 		},
 	}
 	cdr := &dcc.deferredCommitRange
+	tr.mu.RLock()
+	dbRound := tr.dbRound
```

Comment:
Once you start holding the lock more than a line, maybe better to use defer.

Reply:
I extended the lock, can't use defer without nesting it into a function b/c the chanel write at the end must be non-locked.

Reply:
I see, thanks.